torch/_refs/__init__.py

```diff
@@ -801,9 +809,61 @@ def add(
     type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.DEFAULT,
 )
 
+
+def _copysign(
+    a: Union[TensorLikeType, NumberType], b: Union[TensorLikeType, NumberType]
+):
+    if isinstance(b, Number) and isinstance(a, Tensor):
+        b = scalar_tensor(b, dtype=a.dtype, device=a.device)
+    elif isinstance(a, Tensor) and isinstance(b, Tensor) and a.device != b.device:
+        if a.device == torch.device("cpu"):
+            msg = "Expected divisor (b) to be on the same device ({0}) as dividend (a), but it is found on {1}!".format(
+                a.device, b.device
+            )
+            raise RuntimeError(msg)
+        else:
+            b = prims.device_put(b, device=a.device)
```

Comment:
This isn't correct, is it?

```
a = torch.randn(2, 2)
b = torch.randn(2, 2).cuda()
torch.copysign(b, a)
```

in PyTorch we try to avoid this kind of implicit data movement, and this would be a bug in copysign.

Does `_make_elementwise_binary_reference` not handle all the scalar and device behavior for copysign?

Reply:
It seems we move argument b if it is a scalar tensor where ndim=0.

```Python
>>> x = torch.tensor([ 3., -2.,  7., -2., -5.], device='cuda:0')
>>> y = torch.tensor(-8.4784)
>>> torch.copysign(x, y)
```

Reply:
That's correct, because those scalar tensors are conceptually equivalent to a regular number, but the prim should accept them without their being on the same device, I think?

Reply:
> I see this exception when `TorchRefsMode.push(strict=True)`:
> 
> ```
>   File "torch/_prims/wrappers.py", line 258, in _fn
>     args_[0] = torch.tensor(args[0], dtype=dtype)
> RuntimeError: no _refs support for torch.tensor
> ```

We can xfail that test for now, it's OK